cosmology/GravityFieldContainer.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef IPPL_FIELD_CONTAINER_H
+#define IPPL_FIELD_CONTAINER_H
+
+#include <memory>
+
+#include "Manager/BaseManager.h"
+
+// Define the FieldsContainer class
+/**
+ * @brief A container class that manages various fields, mesh, and field layouts
+ *
+ * @tparam T Data type used for the fields (e.g., double)
+ * @tparam Dim Dimensionality of the container (default is 3)
+ */
+template <typename T, unsigned Dim = 3>
+class FieldContainer{
+public:
+
+    /**
+     * @brief Constructor for the FieldContainer class
+     * 
+     * @param hr Grid spacing vector
+     * @param rmin Minimum range vector
+     * @param rmax Maximum range vector
+     * @param decomp Decomposition array indicating which dimensions are decomposed
```

Comment:
I think "...which dimensions are distributed among MPI ranks" is a clearer description